src/java/org/apache/cassandra/db/streaming/CassandraStreamReceiver.java

```diff
@@ -242,6 +246,19 @@ public void finished()
             }
             else
             {
+                if (receivedEntireSSTable)
```

Comment:
I could add a bit more inline commentary to this, but when we don't stream the entire SSTable, validation doesn't make much sense, as the SAI components have just been successfully written when we arrive here. Any error doing that would have already aborted the streaming txn.